packages/astro/test/ssr-prerender-get-static-paths.test.js

```diff
@@ -1,131 +1,282 @@
+					expect(res.status).to.equal(200);
+				});
 
-		res = await fixture.fetch('/c');
-		expect(res.status).to.equal(200);
-	});
+				it('resolves 404 on pattern match without static path - rest params', async () => {
+					const res = await fixture.fetch('/blog/pizza/pizza-hut');
+					const html = await res.text();
 
-	describe('404 behavior', () => {
-		it('resolves 200 on matching static path - named params', async () => {
-			const res = await fixture.fetch('/pizza/provolone-sausage');
-			expect(res.status).to.equal(200);
-		});
+					expect(res.status).to.equal(404);
+					expect(html).to.match(/404/);
+				});
+			});
 
-		it('resolves 404 on pattern match without static path - named params', async () => {
-			const res = await fixture.fetch('/pizza/provolone-pineapple');
-			const html = await res.text();
-			expect(res.status).to.equal(404);
-			expect(html).to.match(/404/);
-		});
+			describe('route params type validation', () => {
+				it('resolves 200 on nested array parameters', async () => {
+					const res = await fixture.fetch('/blog/nested-arrays/slug1');
+					expect(res.status).to.equal(200);
+				});
 
-		it('resolves 200 on matching static path - rest params', async () => {
-			const res = await fixture.fetch('/pizza/grimaldis/new-york');
-			expect(res.status).to.equal(200);
-		});
+				it('resolves 200 on matching static path - string params', async () => {
+					// route provided with { params: { year: "2022", slug: "post-2" }}
+					const res = await fixture.fetch('/blog/blog/2022/post-1');
+					expect(res.status).to.equal(200);
+				});
+
+				it('resolves 200 on matching static path - numeric params', async () => {
+					// route provided with { params: { year: 2022, slug: "post-2" }}
+					const res = await fixture.fetch('/blog/blog/2022/post-2');
+					expect(res.status).to.equal(200);
+				});
+			});
+
+			it('resolves 200 on matching static paths', async () => {
+				// routes params provided for pages /posts/1, /posts/2, and /posts/3
+				for (const page of [1, 2, 3]) {
+					let res = await fixture.fetch(`/blog/posts/${page}`);
+					expect(res.status).to.equal(200);
 
-		it('resolves 404 on pattern match without static path - rest params', async () => {
-			const res = await fixture.fetch('/pizza/pizza-hut');
-			const html = await res.text();
-			expect(res.status).to.equal(404);
-			expect(html).to.match(/404/);
+					const html = await res.text();
+					const $ = cheerio.load(html);
+
+					const canonical = $('link[rel=canonical]');
+					expect(canonical.attr('href')).to.equal(
+						`https://mysite.dev/blog/posts/${page}`,
+						`doesn't trim the /${page} route param`
+					);
+				}
+			});
 		});
 	});
 
-	describe('route params type validation', () => {
-		it('resolves 200 on nested array parameters', async () => {
-			const res = await fixture.fetch('/nested-arrays/slug1');
-			expect(res.status).to.equal(200);
-		});
+	describe('output: "hybrid"', () => {
```

Comment:
Same tests as the above. Refactoring them to reduce code length will result in less readable tests